Windows tests fail with long branch names due to MAX_PATH issues

Example:

* https://github.com/dotnet/macios/pull/22630#issuecomment-2830007459

```
Xamarin.Tests.DotNetProjectTest.IsNotMacBuild(MacCatalyst,"maccatalyst-arm64;maccatalyst-x64",False) (Failed)
'dotnet build' failed with exit code 1
Full command: c:\AzDO\_work\2\s\macios\tests\dotnet\Windows\bin\dotnet\dotnet.exe build "c:\\AzDO\\_work\\2\\s\\macios\\tests\\dotnet\\MySimpleApp\\MacCatalyst\\MySimpleApp.csproj" /p:_BundlerVerbosity=1 "/p:RuntimeIdentifiers=\"maccatalyst-arm64;maccatalyst-x64\"" /p:IsMacEnabled=false /p:Configuration=Debug "/bl:c:\\AzDO\\_work\\2\\s\\macios\\tests\\dotnet\\MySimpleApp\\MacCatalyst\\log-build-20250425_020125.binlog" /v:diag /consoleloggerparameters:Verbosity=Quiet
Listing first 2 error(s) (of 2 error(s)):
    C:\AzDO\_work\2\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\9.0.106\Microsoft.Common.CurrentVersion.targets(5035): error MSB3030: Could not copy the file "C:\AzDO\_work\2\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.MacCatalyst.Runtime.maccatalyst-arm64.net9.0_18.4\18.4.9349-ci.dev-rolf-issue-22610-corebluetooth-scanfornullperipherals\runtimes\maccatalyst-arm64\native\Microsoft.MacCatalyst.registrar.a" because it was not found.
    C:\AzDO\_work\2\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\9.0.106\Microsoft.Common.CurrentVersion.targets(5035): error MSB3030: Could not copy the file "C:\AzDO\_work\2\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.MacCatalyst.Runtime.maccatalyst-arm64.net9.0_18.4\18.4.9349-ci.dev-rolf-issue-22610-corebluetooth-scanfornullperipherals\runtimes\maccatalyst-arm64\native\libxamarin-dotnet-nativeaot-debug.a" because it was not found.
```

* https://github.com/dotnet/macios/pull/22679#issuecomment-2840793872

```
Xamarin.Tests.WindowsTest.PluralRuntimeIdentifiersWithRemoteMac(iOS,"iossimulator-arm64;iossimulator-x64") (Failed)
'dotnet build' timed out after 00:10:00
Full command: D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\dotnet.exe build "D:\\AzDO\\_work\\14\\s\\macios\\tests\\dotnet\\MySimpleApp\\iOS\\MySimpleApp.csproj" /p:_BundlerVerbosity=1 /p:ServerAddress=10.64.38.178 /p:ServerUser=builder /p:ServerPassword=12Ab_Cd3423Bc_De45 /p:ContinueOnDisconnected=false "/p:RuntimeIdentifiers=\"iossimulator-arm64;iossimulator-x64\"" "/bl:D:\\AzDO\\_work\\14\\s\\macios\\tests\\dotnet\\MySimpleApp\\iOS\\log-build-20250429_200958.binlog" /v:diag /consoleloggerparameters:Verbosity=Quiet
Listing first 5 error(s) (of 5 error(s)):
    D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\10.0.100-preview.4.25229.7\Microsoft.Common.CurrentVersion.targets(5094): error MSB3030: Could not copy the file "D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.iOS.Runtime.iossimulator-arm64.net10.0_18.4\18.4.10478-ci.darc-release-10-0-1xx-preview4-7a80aac9-05c6-4446-805a-f95403e8e21b\runtimes\iossimulator-arm64\native\Microsoft.iOS.registrar.a" because it was not found.
    D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\10.0.100-preview.4.25229.7\Microsoft.Common.CurrentVersion.targets(5094): error MSB3030: Could not copy the file "D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.iOS.Runtime.iossimulator-arm64.net10.0_18.4\18.4.10478-ci.darc-release-10-0-1xx-preview4-7a80aac9-05c6-4446-805a-f95403e8e21b\runtimes\iossimulator-arm64\native\libxamarin-dotnet-debug.a" because it was not found.
    D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\10.0.100-preview.4.25229.7\Microsoft.Common.CurrentVersion.targets(5094): error MSB3030: Could not copy the file "D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.iOS.Runtime.iossimulator-arm64.net10.0_18.4\18.4.10478-ci.darc-release-10-0-1xx-preview4-7a80aac9-05c6-4446-805a-f95403e8e21b\runtimes\iossimulator-arm64\native\libxamarin-dotnet-debug.dylib" because it was not found.
    D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\10.0.100-preview.4.25229.7\Microsoft.Common.CurrentVersion.targets(5094): error MSB3030: Could not copy the file "D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.iOS.Runtime.iossimulator-arm64.net10.0_18.4\18.4.10478-ci.darc-release-10-0-1xx-preview4-7a80aac9-05c6-4446-805a-f95403e8e21b\runtimes\iossimulator-arm64\native\libxamarin-dotnet-nativeaot-debug.a" because it was not found.
    D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\sdk\10.0.100-preview.4.25229.7\Microsoft.Common.CurrentVersion.targets(5094): error MSB3030: Could not copy the file "D:\AzDO\_work\14\s\macios\tests\dotnet\Windows\bin\dotnet\packs\Microsoft.iOS.Runtime.iossimulator-arm64.net10.0_18.4\18.4.10478-ci.darc-release-10-0-1xx-preview4-7a80aac9-05c6-4446-805a-f95403e8e21b\runtimes\iossimulator-arm64\native\libxamarin-dotnet-nativeaot.a" because it was not found.
```

--- a/tools/devops/automation/templates/windows/build.yml
+++ b/tools/devops/automation/templates/windows/build.yml
@@ -166,6 +166,15 @@ steps:
     }
   displayName: "Cleanup temporary files"
 
+- pwsh: |
+    Write-Host "Current long paths state:"
+    Get-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem -Name LongPathsEnabled
+    Write-Host "Enabling long paths":
+    Set-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem -Name LongPathsEnabled -Value 1
+    Write-Host "Updated long paths state:"
+    Get-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem -Name LongPathsEnabled
+  displayName: 'Enable long paths'
+
 # This task fixes errors such as these:
 #     error MSB4242: SDK Resolver Failure: "The SDK resolver "NuGetSdkResolver" failed while attempting to resolve the SDK "Microsoft.Build.NoTargets/3.3.0".
 #                    Exception: "NuGet.Packaging.Core.PackagingException: Unable to find fallback package folder 'D:\Program Files (x86)\Microsoft Visual Studio\Shared\NuGetPackages'.

